dandischema/digests/zarr.py

```diff
@@ -15,15 +15,16 @@ class ZarrChecksum(pydantic.BaseModel):
     """
     A checksum for a single file/directory in a zarr file.
 
-    Every file and directory in a zarr archive has a path and a MD5 hash.
+    Every file and directory in a zarr archive has a name and a MD5 hash.
     """
 
     md5: str
-    path: str
+    name: str
+    size: int
```

Comment:
I wonder if to simplify transition there could be a proper deprecation cycle which would support path, and decompose it into name and size as needed. Should be simple right? Wdyt?

Reply:
I don't think that's necessary. As soon as this is released and incorporated in dandi-archive and dandi-cli, we can just recompute checksums on all zarrs and `path` will be updated to `name`.

Reply:
well, since there is no way to announce that `dandischema` (next release) breaks `dandi-cli << (whatever would become compatible with new interface)` people would end up with a dandi-cli which might puke an exception if they manage to upgrade dandischema but not dandi-cli.  unlikely but possible.  deprecation cycle allows to amortize perturbation and avoid such problems...
oh well -- since python is dynamic and zarr isn't primary target yet for users, I am ok to proceed without deprecation cycle at hopefully low or none count of affected users.

Reply:
If we make the next dandischema release a new minor version, it would be incompatible with dandi-cli's `dandischema ~= 0.5.1` requirement and thus help to prevent that from happening (but there would still be problems with checksum failures if people use the old client to upload Zarrs).

Reply:
ah, right, nice.
@jwodder Could you prep a PR for dandi-cli to accompany this one?